pupil_src/player/vis_circle.py

```diff
@@ -42,9 +42,22 @@ def update(self,frame,events):
         else:
             thickness = self.thickness
 
-        pts = [denormalize(pt['norm_pos'],frame.img.shape[:-1][::-1],flip_y=True) for pt in events.get('gaze_positions',[])]
-        for pt in pts:
-            transparent_circle(frame.img, pt, radius=self.radius, color=(self.b, self.g, self.r, self.a), thickness=thickness)
+        pts = [(
```

Comment:
It was actually a debugging feature which appeared useful afterwards. The idea is that pupil positions which are part of a fixation are drawn red. If there are no fixations (or the fixation detector is disabled) the points are drawn as before, i.e. it does not assume a working fixation detector.

Therefore I would recommend to add this feature to the other visualizers as well instead of creating a separate one.